postgis/lwgeom_remove_small_parts.c

```diff
@@ -0,0 +1,244 @@
+/**********************************************************************
+ *
+ * PostGIS - Spatial Types for PostgreSQL
+ * http://postgis.net
+ *
+ * PostGIS is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * PostGIS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PostGIS.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **********************************************************************
+ *
+ * Copyright (C) 2024 Sam Peters <gluser1357@gmx.de>
+ *
+ **********************************************************************/
+
+#include "postgres.h"
+#include "funcapi.h"
+#include "utils/array.h"
+#include "utils/builtins.h"
+#include "utils/lsyscache.h"
+#include "utils/numeric.h"
+#include "access/htup_details.h"
+
+#include "liblwgeom.h"
+#include "liblwgeom_internal.h"
+
+// ===============================================================================
+// helper function for polygon and polyline POINTARRAY's
+// which removes small parts given by dx and dy.
+// ===============================================================================
+static void ptarray_remove_dim_helper(POINTARRAY *points, double mindx, double mindy) {
+
+    int r;
+    double xmin = 0, ymin = 0, xmax = 0, ymax = 0;
+    double x, y;
+    POINT4D point;
+
+    int npoints = points->npoints;
+    for (r=0; r < npoints; r++) {
+
+		getPoint4d_p(points, r, &point); // point read/write see ptarray_flip_coordinates
+
+		x = point.x;
+		y = point.y;
+
+		if (mindx > 0) {
```

Comment:
Nit:

`mindx` and `mindy` are guaranteed to be `> 0` because of https://github.com/postgis/postgis/pull/768/files#diff-601ccecc3a28db07fae7003a697253347d840a705e7536e1cd25d6d2edf28169R112, so an assert or check and early return from `ptarray_remove_dim_helper` would make the code simpler.

Reply:
Either mindx or mindy can be <= 0, but not both. IMO this check is required and ok?